src/db.ts

```diff
@@ -461,46 +479,39 @@ export const saveCardReference = async (cardRef) => {
 
   const cardReferences = await getCardReferences();
   cardReferences.push(cardRef);
-  await redisClient.setAsync(
+  await redisClient.set(
     `${process.env.MOCKSOLARIS_REDIS_PREFIX}:cardReferences`,
     JSON.stringify(cardReferences)
   );
 
   return true;
 };
 
-export const getCardData = async (cardId) => {
-  const persons = await getAllPersons();
-
-  const cardData = _(persons)
-    .map((person) => _.get(person, "account.cards", []))
-    .flatten()
-    .value()
-    .find((cd) => cd.card.id === cardId);
-
-  return cardData;
+export const getCardData = async (cardId: string): Promise<Card> => {
+  const personWhoOwnsTheCard = await findPerson(
+    (p) => !!(p?.account?.cards || []).find((cd) => cd?.card?.id === cardId)
+  );
+  return personWhoOwnsTheCard?.account?.cards.find(
+    (card: CardData) => card?.card?.id === cardId
+  );
 };
 
 export const getPersonBySpendingLimitId = async (id) => {
-  const persons = await getAllPersons();
-
-  const cardData = _(persons)
-    .map((person) => _.get(person, "account.cards", []))
-    .flatten()
-    .value()
-    .find(({ controls }) => controls?.find((control) => control.id === id));
-
-  return { person: await getPerson(cardData.card.person_id), cardData };
+  const person = await findPerson((p) => {
+    return !!(p.account?.cards ?? []).find(
+      (c) => !!(c.controls ?? []).find((co) => co.id === id)
+    );
+  });
+  const cardData = (person?.account?.cards ?? []).find(
+    (c) => !!(c.controls ?? []).find((co) => co.id === id)
+  );
```

Comment:
:question: is there a better way to write this? `(c) => !!(c.controls ?? []).find((co) => co.id === id)` was repeated for no reason.

Reply:
This commit makes them non-duplicates: https://github.com/kontist/mock-solaris/pull/114/commits/bc4e4071f154373cfe2a99b139caf81b9af58520